upf/upf_api.c

```diff
@@ -373,6 +375,103 @@ static void vl_api_upf_nat_pool_dump_t_handler
   }
 }
 
+/* API message handler */
+static void
+vl_api_upf_policy_add_del_t_handler (vl_api_upf_policy_add_del_t * mp)
+{
+  vl_api_upf_policy_add_del_reply_t *rmp = NULL;
+  upf_main_t *sm = &upf_main;
+  fib_route_path_t *rpaths = NULL, *rpath;
+  vl_api_fib_path_t *apath;
+  u8 *policy_id = 0;
+  u8 action = mp->action;
+  int ii = 0;
+  int rv = 0;
+
+  /* Make sure ID is null terminated */
+  mp->identifier[sizeof (mp->identifier) - 1] = 0;
+
+  policy_id = format (0, "%s", mp->identifier);
+
+  if (0 != mp->n_paths)
+    vec_validate (rpaths, mp->n_paths - 1);
+
+  for (ii = 0; ii < mp->n_paths; ii++)
+    {
+      apath = &mp->paths[ii];
+      rpath = &rpaths[ii];
+
+      rv = fib_api_path_decode (apath, rpath);
+
+      if (0 != rv)
+	goto out;
```

Comment:
is there no reply at all in case of an error?

Reply:
Autoreply won't be sent so bianpi client (upg-manager) won't receive a response and convert it into internal error. This will be a trigger if something went wrong